src/main/java/org/minions/devfund/katerinaanzoleaga/battleship/Ship.java

```diff
@@ -0,0 +1,278 @@
+     * @return int
+     */
+    public int getBowColumn() {
+        return bowColumn;
+    }
+
+    /**
+     * Returns the length of the ship.
+     * @return length
+     */
+    public int getLength() {
+        return length;
+    }
+
+    /**
+     * Sets the lenght of the ship from extended classes.
+     * @param length int
+     */
+    public void setLength(int length) {
+        this.length = length;
+    }
+
+    /**
+     * Returns true if the ship is horizontal.
+     * @return bool
+     */
+    public boolean isHorizontal() {
+        return horizontal;
+    }
+
+    /**
+     * Retruns the hits array.
+     * @return hits array
+     */
+    public boolean[] getHit() {
+        return hit.clone();
+    }
+
+    /**
+     * Updates the hit array.
+     * @param hit hits array
+     */
+    protected void setHit(final boolean[] hit) {
+        this.hit = hit.clone();
+    }
+
+
+    /**
+     * Abstract method to be overwrite by the extended classes.
+     * @return string
+     */
+    abstract String getShipType();
+
+
+    /**
+     * Given the starting position, either row or column, this fuction returns.
+     * true if the ship's larger position will feit in the ocean.
+     * @param position either row or column
+     * @param ocean the ocean reference.
+     * @return ture if fits.
+     */
+    private boolean fitsInDirection(int position, final Ocean ocean) {
+        if (position >= 0) {
+            return position + this.getLength() - 1 < ocean.getShipArray().length;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Given the current ship.
+     * @param row int
+     * @param column int
+     * @param horizontal bool
+     * @param ocean ocean
+     * @return True if it is possible to place a ship in the defined position.
+     */
+    public boolean okToPlaceShipAt(int row, int column, boolean horizontal, final Ocean ocean) {
+        if (ocean.isValidPosition(row, column)) {
+            if (horizontal) {
+                if (fitsInDirection(column, ocean)) {
+                    for (int i = row - 1; i <= row + 1; ++i) {
+                        for (int j = column - 1; j <= column + this.getLength(); ++j) {
+                            if (ocean.isValidPosition(i, j)) {
+                                if (ocean.isOccupied(i, j)) {
+                                    return false;
+                                }
+                            }
+                        }
+                    }
+                    return true;
+                } else {
+                    return false;
+                }
+
+            } else {
+                if (fitsInDirection(row, ocean)) {
+                    for (int i = row - 1; i <= row + this.getLength(); ++i) {
+                        for (int j = column - 1; j <= column + 1; ++j) {
+                            if (ocean.isValidPosition(i, j)) {
+                                if (ocean.isOccupied(i, j)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)